db/db_impl.cc

```diff
@@ -912,6 +913,25 @@ Status DBImpl::SetDBOptions(
     InstrumentedMutexLock l(&mutex_);
     s = GetMutableDBOptionsFromStrings(mutable_db_options_, options_map,
                                        &new_options);
+    if (new_options.bytes_per_sync == 0) {
+      new_options.bytes_per_sync = 1024 * 1024;
```

Comment:
Curious, why change it from the place where it was earlier? Why does this need to be done before `BuildDBOptions` ?

Reply:
Previously there was only one BuildDBOptions at line 964. Now that we have two, I optimized it for the 2nd to use the result of the 1st. For the however there should not be any change between the two. Updating bytes_per_sync was the only such change, so I moved it before the 1st BuildDBOptions.

Reply:
Where's the second BuildDBOptions? I see only one on line 920. Maybe I missed the other one.

Reply:
https://github.com/facebook/rocksdb/pull/5368/files#diff-71a34e2e4e097b44064da7d3a2b68bd8L964
It is removed now.

Reply:
Oh, I thought you were saying there are two calls to BuildDBOptions in the new code.